tests/core/test_actions.py

```diff
@@ -344,6 +362,156 @@ async def test_remote_action_with_template_param(
     ]
 
 
+@pytest.mark.parametrize("event_class", rasa.shared.utils.common.all_subclasses(Event))
+async def test_remote_action_validate_all_event_subclasses(event_class: Type[Event]):
+    response = {"events": [{"event": event_class.type_name}], "responses": []}
```

Comment:
would it be possible to create objects out of this classes and then call `as_dict()`? This test doesn't seem particularly realistic (nobody would send an event which only has the `event` key)

Reply:
It depends what the purpose is: I thought this test only checks that any new potential event is added to the schema as well, otherwise it fails if forgotten. 
The test below this one `test_remote_action_valid_payload_all_events` verifies that the event objects are valid - I can tweak those to add more arguments for a more realistic scenario?

Reply:
The advantage of `as_dict` would be that we'd have relatively realistic payloads which automatically change when somebody does changes to the `as_dict` implementation. In my opinion some of these events shouldn't be valid if just the `event` key is provided.

Reply:
Yes, some of them failed after enforcing required properties - I've hacked the test for now 😅  but my problem was how to instantiate an event from class type? Do I use static method `from_parameters`? Although the if/else will get very long since parameters will differ event by event. If that works, I can use `as_dict` then in the response.

Reply:
mhm 😬 Note sure, but do all Event classes have defaults for all constructor params? If not I'd simply add a test which makes sure that each `Event.type_name` is present in the schema and then hope that you're test below this captures the rest 👍🏻